code/game/objects/items/weapons/melee/energy.dm

```diff
@@ -176,36 +131,50 @@
 	icon_state = "esaw_0"
 	icon_state_on = "esaw_1"
 	hitcost = 75 //Costs more than a standard cyborg esword
-	item_color = null
 	w_class = WEIGHT_CLASS_NORMAL
 	sharpness = IS_SHARP
 	light_color = "#40ceff"
-	possible_colors = null
 
-/obj/item/weapon/melee/energy/sword/cyborg/saw/Initialize()
-	. = ..()
-	icon_state = "esaw_0"
-	item_color = null
-
-/obj/item/weapon/melee/energy/sword/cyborg/saw/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+/obj/item/weapon/melee/transforming/energy/sword/cyborg/saw/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
 	return 0
 
-/obj/item/weapon/melee/energy/sword/saber
+/obj/item/weapon/melee/transforming/energy/sword/saber
+	var/list/possible_colors = list("red", "blue", "green", "purple")

```

Comment:
This could be a be an assoc with `"color" = color define`, or even just the color define if you don't have to use the string for anything.